### PR TITLE
Add 'allow-non-simd' feature and fix write-str-simd conditional compilation

### LINKED ISSUE #26: cannot find function `write_str_simd` in this scope
crate errors out when compiling to `armv7-unknown-linux-gnueabihf` with the following error:

```
error[E0425]: cannot find function `write_str_simd` in this scope
Error:    --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/value-trait-0.4.0/src/generator.rs:162:19
    |
162 |             stry!(write_str_simd(self.get_writer(), &mut string,));
    |                   ^^^^^^^^^^^^^^ not found in this scope
```

### PR DESCRIPTION
**Pull request by mfelsche**
This should fix #26

### CODE REVIEW

**Comment by mfelsche** [comment]
Compiling this branch on arm:

```
$ cross build --target=arm-unknown-linux-gnueabihf

   Compiling value-trait v0.5.1 (/project)
error: write_str_simd not supported on the current architecture
   --> src/generator.rs:370:9
    |
370 |         compile_error!("write_str_simd not supported on the current architecture")
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `value-trait` due to previous error
```


Compiling this branch on arm with `allow-non-simd`:
```
$ cross build --target=arm-unknown-linux-gnueabihf --features="allow-non-simd"

   Compiling value-trait v0.5.1 (/project)
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
```

File: src/generator.rs [hunk 1]
```diff
@@ -316,10 +283,7 @@ pub trait BaseGenerator {
         Ok(())
     }
 
-    #[cfg(all(
-        any(target_arch = "x86", target_arch = "x86_64"),
-        not(target_feature = "avx2")
-    ))]
+    #[cfg(target_feature = "sse2")]
```
Comment (Licenser) [inline src/generator.rs:286]: should the guard here include `not(target_feature = "avx2")` to ensure that we don't include both implementations when we have a sse2 and avx2 compatible CPU?

**Review by Licenser — APPROVED** [review]
You're the best 🚀

**Comment by hustcer** [comment]
I got the same error while building `armv7` and `riscv64` version of Nushell packages, detailed error could be found here: https://github.com/nushell/nushell/actions/runs/5239108268/jobs/9458637974, Is there a work around? Thanks

**Pull request merged**